amethyst_renderer/src/pass/debug_lines/interleaved.rs

```diff
@@ -0,0 +1,138 @@
+//! Debug lines pass
+
+use super::*;
+use amethyst_core::cgmath::{Matrix4, One};
+use amethyst_core::specs::{Join, Read, ReadStorage, Write, WriteStorage};
+use amethyst_core::transform::GlobalTransform;
+use cam::{ActiveCamera, Camera};
+use debug_drawing::{DebugLine, DebugLines, DebugLinesComponent};
+use error::Result;
+use gfx::pso::buffer::ElemStride;
+use gfx::Primitive;
+use gfx_core::state::{Blend, ColorMask};
+use mesh::Mesh;
+use pass::util::{get_camera, set_attribute_buffers, set_vertex_args, setup_vertex_args};
+use pipe::pass::{Pass, PassData};
+use pipe::{DepthMode, Effect, NewEffect};
+use std::marker::PhantomData;
+use types::{Encoder, Factory};
+use vertex::{Color, Normal, Position, Query};
+
+/// Draw several simple lines for debugging
+///
+/// See the [crate level documentation](index.html) for information about interleaved and separate
+/// passes.
+///
+/// # Type Parameters:
+///
+/// * `V`: `VertexFormat`
+#[derive(Derivative, Clone, Debug, PartialEq)]
+#[derivative(Default(bound = "V: Query<(Position, Color, Normal)>"))]
+pub struct DrawDebugLines<V> {
+    _pd: PhantomData<V>,
+    transparency: Option<(ColorMask, Blend, Option<DepthMode>)>,
+}
+
+impl<V> DrawDebugLines<V>
+where
+    V: Query<(Position, Color, Normal)>,
+{
+    /// Create instance of `DrawDebugLines` pass
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Enable transparency
+    pub fn with_transparency(
```

Comment:
Should we even have this method if it is unimplemented? If so we might want a comment in the docs about it.

Reply:
Probably it's better removed, so far I haven't been able to make it work at all and it's not a lot of code.